packages/graphback-datasync/src/providers/DataSyncConflictProvider.ts

```diff
@@ -0,0 +1,126 @@
+import { ModelDefinition, GraphbackContext, TransformType, FieldTransform, NoDataError, GraphbackOperationType } from '@graphback/core';
+import { DataSyncModelConflictConfig, DataSyncFieldNames, ConflictMetadata, ConflictError } from '../util';
+import { MongoDeltaHelper } from '../deltaHelper';
+import { DataSyncMongoDBDataProvider } from './DatasyncMongoDBDataProvider';
+
+export const MAX_RETRIES = 3;
+
+/**
+ * Da
+ */
+export class DataSyncConflictMongoDBDataProvider<Type = any> extends DataSyncMongoDBDataProvider<Type> {
+  protected conflictConfig: DataSyncModelConflictConfig
+  protected deltaHelper: MongoDeltaHelper
+
+  public constructor(model: ModelDefinition, client: any, dataSyncConflictConfig: DataSyncModelConflictConfig) {
+    super(model, client);
+    this.conflictConfig = dataSyncConflictConfig;
+
+    if (!this.conflictConfig.conflictResolution?.resolveUpdate) {
+      this.conflictConfig.conflictResolution = {
+        resolveUpdate(conflict: ConflictMetadata) {
+          throw new ConflictError(conflict)
+        }
+      }
+    }
+
+    this.deltaHelper = new MongoDeltaHelper(model, client);
+
+  }
+
+  public async create(data: any, context: GraphbackContext): Promise<Type> {
+    const contextWithoutSelectedFields: GraphbackContext = { ...context }
+    contextWithoutSelectedFields.graphback.options.selectedFields = [];
+
+    data[DataSyncFieldNames.version] = 1;
+
+    const result = await super.create(data, contextWithoutSelectedFields);
+
+    await this.deltaHelper.insertDiff(result).catch((e: any) => { console.error(`Error in inserting delta: ${e}`) });
+
+
+    return result;
+  }
+
+  public async update(updateDocument: any, context: GraphbackContext): Promise<Type> {
+    const { _id } = updateDocument;
+
+    for(let i = 0; i < MAX_RETRIES; i++) {
+      
+      const serverData = await this.db.collection(this.collectionName).findOne({ _id, [DataSyncFieldNames.deleted]: false});
+      const base = await this.deltaHelper.findBaseForConflicts(updateDocument);
+      const clientData = updateDocument;
+
+      let resolvedUpdate = Object.assign({}, updateDocument);
+
+      const updateFilter: any = { _id, [DataSyncFieldNames.version]: serverData[DataSyncFieldNames.version] }
```

Comment:
Explicit type declarations are great, but when it is `any` there is no need to do it.

```suggestion
      const updateFilter = { _id, [DataSyncFieldNames.version]: serverData[DataSyncFieldNames.version] }
```